src/search.c

```diff
@@ -0,0 +1,123 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+#include <stddef.h>
+#include "search.h"
+
+int find_match(char* line, char* word, int pos_start, int lineNum, list_t* matches)
+{
+    if (strlen(word) > strlen(line)) return -1;
+
+    char* dup = strdup(line);
+
+    char* line2 = strdup(line);
+    int pos = pos_start;
+    char* token = strtok_r(line2, " ,.!?\r\t\n", &line2);
+    match* foundMatch = NULL;
+
+    while (token != NULL) {
+
+        if (strcmp(token, word) == 0) {
```

Comment:
All uses of `strcmp` need to be uses of `strncmp`

Reply:
Done!